File: mavkit-python/examples/monitor_statustext.py
"""Print STATUSTEXT messages from the vehicle (Ctrl+C to stop)."""

import asyncio
import os

import mavkit


async def main():
    bind_addr = os.environ.get("MAVKIT_EXAMPLE_UDP_BIND", "0.0.0.0:14550")

    vehicle = await mavkit.Vehicle.connect_udp(bind_addr)
    status_text = vehicle.telemetry().messages().status_text()
    print("listening for status messages (Ctrl+C to stop)...\n")

    while True:
        event = (await status_text.wait()).value
        print(f"[{event.severity}] {event.text}")


asyncio.run(main())

File: examples/monitor_statustext.rs
use mavkit::Vehicle;

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error>> {
    let bind_addr =
        std::env::var("MAVKIT_EXAMPLE_UDP_BIND").unwrap_or_else(|_| "0.0.0.0:14550".to_string());

    let vehicle = Vehicle::connect_udp(&bind_addr).await?;
    println!("listening for status messages (Ctrl+C to stop)...\n");

    let mut st_rx = vehicle.statustext();

    loop {
        st_rx.changed().await?;
        if let Some(msg) = st_rx.borrow().clone() {
            println!("[{}] {}", msg.severity, msg.text);
        }
    }
}
